graphql_compiler/compiler/emit_sql.py

```diff
@@ -590,21 +629,25 @@ def add_filter(self, predicate, aliases):
         sql_expression = predicate.to_sql(self._dialect, aliases, self._output_vertex_alias)
         self._filters.append(sql_expression)
 
-    def end_fold(self, alias_generator, from_clause, outer_from_table):
+    def end_fold(
+        self,
+        alias_generator: "UniqueAliasGenerator",
+        from_clause: FromClause,
+        outer_from_table: Alias,
+    ) -> Tuple[Select, Join, FoldScopeLocation]:
         """Produce the final subquery and join it onto the rest of the query."""
         if self._ended:
             raise AssertionError(
                 "Cannot call end_fold more than once. "
-                "Invalid state encountered during fold {}".format(self)
+                f"Invalid state encountered during fold {self}."
             )
-        if self._output_vertex_alias is None:
+        if self._output_vertex_alias is None or self._output_vertex_location is None:
             raise AssertionError(
-                "No output vertex visited. Invalid state encountered during fold {}".format(self)
+                f"No output vertex visited. Invalid state encountered during fold {self}."
             )
         if len(self._traversal_descriptors) == 0:
             raise AssertionError(
-                "No traversed vertices visited. "
-                "Invalid state encountered during fold {}".format(self)
+                "No traversed vertices visited. " f"Invalid state encountered during fold {self}."
```

Comment:
This seems like a `black`-ism :) Perhaps merge the two strings into one f-string?